custom_components/sat/minimum_setpoint.py

```
@@ -1,87 +1,414 @@
-        self._relative_modulation_level = update_state(
-            tolerance=0.01,
-            previous=self._relative_modulation_level,
-            new_value=boiler_state.relative_modulation_level,
-        )
+            regime_state.minimum_setpoint += self._config.increase_step
+            regime_state.minimum_setpoint = self._clamp_setpoint(regime_state.minimum_setpoint)
+            _LOGGER.debug(f"Updated regime minimum_setpoint=%.1f after starvation.", regime_state.minimum_setpoint)
 
-        if self._base_return_temperature.value is None:
-            _LOGGER.debug("Skip calculation: base return temperature is not set.")
+    def on_cycle_end(self, boiler_state: BoilerState, cycles: CycleStatistics, last_cycle: Cycle, requested_setpoint: Optional[float], outside_temperature: Optional[float]) -> None:
```

Pylint (4.0.4)

[refactor] Too many arguments (6/5)

(R0913)